addons/website_event/models/event_event.py

```diff
@@ -102,6 +111,22 @@ def _default_question_ids(self):
                                             domain=[('once_per_order', '=', False)])
 
     def _compute_is_participating(self):
+        events = self._get_events_where_user_participates()
+        for event in self:
+            event.is_participating = event in events
+
+    @api.model
+    def _search_is_participating(self, operator, value):
+        if operator not in ['=', '!=']:
+            raise NotImplementedError(_('This operator is not supported'))
+        if not isinstance(value, bool):
+            raise UserError(_('Value should be True or False (not %)', value))
+        check_has_access = operator == '=' and value or operator == '!=' and not value
```

Comment:
"check_has_access" is a bit of a strange name, why not simply "check_is_participating" ?